docs/CppMetricsApiSdkDesign.md

```diff
@@ -0,0 +1,1105 @@
+
+```
+
+class StdoutExporter
+{
+
+    ExportResult Export(export.CheckpointSet checkpointSet) noexcept
+    {
+    
+       if(checkpointSet != NULL) {
+            
+            ForEach(checkpoint in checkpointSet) {
+                Aggreagtor agg = checkpoint.record.Aggregator();
+                
+                // Check what type of Aggregator it is
+                if(agg.type == Sum) {
+                    cout << agg.Sum() << endl;
+                }
+            }
+       }
+
+
+        return ExportResult::kSuccess;
+    }
+    
+    void shutdown() {
+    try {
+    
+      managedChannel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+      
+    } catch (InterruptedException e) {
+    
+      logger.log(Level.WARNING, "Failed to shutdown the gRPC channel", e);
+      
+    }
+  }
+
+ };
+```
+
+## Test Strategy / Plan
+
+Since there is a specification we will be following, we will not have to write out user stories for testing. We will generally only be writing functional unit tests for this project. The C++ Open Telemetry repository uses [Googletest](https://github.com/google/googletest) because it provides test coverage reports, also allows us to easily integrate code coverage tools such as [codecov.io](http://codecov.io/) with the project. A required coverage target of 90% will help to ensure that our code is fully tested.
+
+An open-source header-only testing framework called [Catch2](https://github.com/catchorg/Catch2) is an alternate option which would satisfy our testing needs. It is easy to use, supports behavior driven development, and does not need to be embedded in the project as source files to operate (unlike Googletest). Code coverage would still be possible using this testing framework but would require us to integrate additional tools such as [gcov](https://gcc.gnu.org/onlinedocs/gcc/Gcov.html) or [lcov](http://ltp.sourceforge.net/coverage/lcov.php). This framework may be preferred as an agnostic replacement for Googletest and is widely used in open source projects.  
+
+## Design Questions
+
+* As the SDK specification is an evolving document, there is no consensus on how the Accumulator class should be implemented. Some languages create a separate class while others fold its functionality into the Meter class.  Which option better suits C++ and the library’s end users?
+* As instruments are injected into user code and often export to remote locations, they create an attack vector which may undermine user applications.  What security considerations should we take when developing this library?  
+* What about design considerations for handling concurrency issues?
+
+
+
+## Contributors
+
+* Alolita Sharma
+* Ankit Bhargava
+* Brandon Kimberly
+* Hudson Humphries
+
+Copyright 2020 [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
```

Comment:
I'm afraid we cannot merge contributions with [such copyright notices](https://github.com/cncf/foundation/blob/master/charter.md#11-ip-policy). We don't have notices consistently over all our files, but if we have them they look like [this](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/api/include/opentelemetry/trace/span_id.h#L1).